benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -265,8 +278,18 @@ private void handleRunningExperimentState(String experimentID)
         break;
       }
 
+      runningState = experimentModelDAO.getRunningState(experimentID);
+      logger.info("handleRunningExperimentState: runningState == " + runningState);
+
       // Exit as soon as final state is executed
-      if (experimentState != null && experimentState == TERMINATED) {
+      if (experimentState == TERMINATED) {
+        exit = true;
+      }
+
+      // Exit while waiting for the Faban Manager to notify about the result of the scheduled
+      // trial. This exits before the execution of HANDLE_EXPERIMENT_RESULT
+      else if (runningState == RunningState.HANDLE_TRIAL_RESULT
```

Comment:
It depends if the system is waiting for input from the `FabanManager` or not. So in the inner loop for running experiment we exit, but in the outer loop we check if we expect a result from the `FabanManager` or not. If we do not expect it we run the loop again so that the `TERMINATED` state handling gets executed.